main.go

```diff
@@ -467,26 +525,35 @@ func check_handler(w http.ResponseWriter, r *http.Request) {
 				text = append(text, scanner.Text())
 			}
 
-			file.Close()
+			err = file.Close()
+			if err != nil {
+				fmt.Println(err)
+			}
 
-			for _, each_ln := range text {
-				if each_ln == tx.PaymentID {
+			for _, eachLn := range text {
+				if eachLn == tx.PaymentID {
 					logged = true
 				}
 			}
+
 			if !logged {
 
 				f, err := os.OpenFile("log/paid.log", os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
 				if err != nil {
 					log.Println(err)
 				}
-				defer f.Close()
+				defer func(f *os.File) {
+					err := f.Close()
+					if err != nil {
+						fmt.Println(err)
+					}
+				}(f)
 				if _, err := f.WriteString(tx.PaymentID + "\n"); err != nil {
 					log.Println(err)
 				}
 
 				c.Meta = ""
-				c.Receipt = string(tx.Amount) + "Payment received! It is safe to close the tab"
+				c.Receipt = strconv.FormatInt(tx.Amount, 10) + "Payment received! It is safe to close the tab"
```

Comment:
Using the strconv library does not affect the binary size in any noticeable way. Still 9.54mb for me.